### DISCUSSION TITLE
Error connexion with vacuum with python plugin domoticz

### DISCUSSION THREAD

**Discussion by Faaab68** [rytilahti/python-miio#943] [Q&A] [answered]
Hello can you help me please?

I follow tutorial Xiaomi Mi Robot Vacuum - Domoticz Python plugin
step by step but i have errors when i put command :

sudo ./miio_server.py 192.168.1.11 xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx --host 127.0.0.1 --port 22222

i have :
pi@raspberrypi:~/domoticz/plugins/xiaomi-mirobot $ sudo ./miio_server.py 192.168.1.11 XXXXXXXXXXXXXXXXXXXXXXXX --host 127.0.0.1 --port 22222
Traceback (most recent call last):
File "./miio_server.py", line 17, in
from miio import Vacuum, DeviceException
File "././.env/lib/python3.7/site-packages/miio/init.py", line 2, in
from miio.protocol import Message, Utils
File "././.env/lib/python3.7/site-packages/miio/protocol.py", line 155, in
Const(Int16ub, 0x2131),
File "././.env/lib/python3.7/site-packages/construct/core.py", line 2504, in init
super(Const, self).init(subcon)
File "././.env/lib/python3.7/site-packages/construct/core.py", line 670, in init
raise TypeError("subcon should be a Construct field")
TypeError: subcon should be a Construct field

Cand you help me please?
Thanks

**Comment by rytilahti** [comment, marked as answer]
I have no clue about that, sorry, the error is not coming from this library. I have seen similar errors related to construct, so if it's possible, try to install version `2.10.56` with `pip install construct=2.10.56` to see if that helps.
